lib/authentification/authentification_base/lib/src/gateways/link_provider_gateway.dart

```diff
@@ -46,6 +46,13 @@ class LinkProviderGateway extends BlocBase {
         EmailAuthProvider.credential(email: email, password: password);
     await userGateway.linkWithCredential(credential);
 
+    // Even when we are using the `userChanges()` stream, we still need to call
+    // `reload()` because the `isAnonymous` property is not updated on iOS (and
+    // macOS). When the bug is fixed, we can remove this call.
+    //
+    // Bug report: https://github.com/firebase/flutterfire/issues/11520
+    await userGateway.reloadUser();
+
```

Comment:
Couldn't the user be reloaded in the `linkWithCredential` method? In this way the code using the `userGateway` could stay "dumber".

Reply:
Yes, that's possible and alos shorter 👍